scipy/optimize/slsqp.py

```diff
@@ -346,30 +393,17 @@ def cjac(x, *args):
     jw = zeros(len_jw)
 
     # Decompose bounds into xl and xu
-    if bounds is None or len(bounds) == 0:
```

Comment:
The diff is a bit confusing here, and being close to the end of the file my mind is starting to check out. What is all this for?

Reply:
SLSQP accepts two arrays for bounds, `xl` and `xu`. Instead of `-np.inf/np.inf` for no bounds on a given parameter it uses `np.nan`. The changes here are a rationalisation of the previous code. `full_new_bounds` (created at the top) already contains float arrays that can be used for `xl` and `xu` (the only thing that needed to be done was to convert any `np.inf` to `np.nan`).